site-src/concepts/api-overview.md

```diff
@@ -133,91 +150,94 @@ different relationships that Gateways and Routes can have:
 
-```
-kind: Gateway
-...
-spec:
-  listeners:
-  - routes:
-      kind: HTTPRoute
-      selector:
-        matchLabels:
-          expose: prod-web-gw
-      namespaces:
-        from: All
-```
+A Route *must* select what Gateway it wants to attach to, based on the
+`parentRefs` field, which allows the selection of the Group, Kind, name,
+and namespace of the object. Cluster-scoped objects can also be selected by
+changing the Scope to `Cluster`. Although only Gateways are currently supported,
+this is intended to allow for later extension. Additionally, the `parentRefs`
+stanza is a list, so a Route may request to attach to more than one Gateway
+(or other parent object).
 
-#### Route Exposure
+Additionally, Gateways *may* specify what kind of Routes they support
+(defaults to Routes that match the Listener protocol if not specified), and
+where those Routes can be (defaults to same namespace). If a Route wants
+to attach to a Gateway in another namespace, that Gateway must *explicitly*
+allow Routes from its namespace for the binding to succeed.
 
-Routes can determine how they are exposed through Gateways. The `gateways.allow`
-field supports three values:
+The Route becomes attached only when the Gateway and Route specifications intersect.
+Note that this means that the binding requires bidirectional agreement between
+the two objects. This is a critical part of the API's role-based structure.
 
-- `All` is the default value if none is specified. This leaves all binding
-to the Route label and Namespace selectors on the Gateway.
-- `SameNamespace` only allows this Route to bind with Gateways from the
-same Namespace.
-- `FromList` allows an explicit list of Gateways to be specifiied that a
-Route will bind with. `gateways.gatewayRefs` is only supported with this option.
+#### Gateway - Route binding examples
 
-The following `my-route` Route selects only the `foo-gateway` in the
-`foo-namespace` and will not be able to bind with any other Gateways. Note that
-`foo-gateway` is in a different Namespace. If the `foo-gateway` allows
-cross-Namespace binding and also selects this Route then `my-route` will bind to
-it.
+The following `my-route` Route wants to attach to the `foo-gateway` in the
+`foo-namespace` and will not bind with any other Gateways. Note that
+`foo-gateway` is in a different Namespace. The `foo-gateway` must allow
+bindings from HTTPRoutes in the namespace `bar-namespace`.
 
 ```yaml
 kind: HTTPRoute
 metadata:
   name: my-route
   namespace: bar-namespace
 spec:
-  gateways:
-    allow: FromList
-    gatewayRefs:
-    - name: foo-gateway
-      namespace: foo-namespace
+  parentRefs:
+  - kind: Gateway
+    name: foo-gateway
+    namespace: foo-namespace
+...
+```
+
+This `foo-gateway` allows the `my-route` HTTPRoute to bind.
+
+```yaml
+kind: Gateway
+metadata:
+  name: foo-gateway
+  namespace: foo-namespace
+spec:
+  listeners:
+  - name: prod-web
+    routes:
+      kinds:
+      - HTTPRoute
```

Comment:
This line is not valid. It should be `kind: HTTPRoute`.
Can we please instead put this into the examples directory and reference it here so that we can ensure that such code segements are valid?